remote/gogs/gogs.go

```diff
@@ -255,3 +270,25 @@ func (c *client) newClientToken(token string) *gogs.Client {
 	}
 	return client
 }
+
+// helper function to return matching hook.
+func matchingHooks(hooks []*gogs.Hook, rawurl string) *gogs.Hook {
```

Comment:
can you add some unit tests for this function similar to
https://github.com/drone/drone/blob/master/remote/bitbucket/bitbucket_test.go#L233